server/conn.go

```diff
@@ -120,14 +121,14 @@ func (cc *clientConn) String() string {
 // handshake works like TCP handshake, but in a higher level, it first writes initial packet to client,
 // during handshake, client and server negotiate compatible features and do authentication.
 // After handshake, client can send sql query to server.
-func (cc *clientConn) handshake() error {
+func (cc *clientConn) handshake(ctx context.Context) error {
 	if err := cc.writeInitialHandshake(); err != nil {
 		return errors.Trace(err)
 	}
-	if err := cc.readOptionalSSLRequestAndHandshakeResponse(); err != nil {
+	if err := cc.readOptionalSSLRequestAndHandshakeResponse(ctx); err != nil {
 		err1 := cc.writeError(err)
 		if err1 != nil {
-			log.Debug(err1)
+			logutil.Logger(ctx).Debug("readOptionalSSLRequestAndHandshakeResponse error", zap.Error(err1))
```

Comment:
should be "writeError failed"? readOptionalSSLRequestAndHandshakeResponse error is recorded in `err`.